tests/performance/scripts/compare.sh

```diff
@@ -444,10 +444,10 @@ create view query_logs as
 create table query_run_metric_arrays engine File(TSV, 'analyze/query-run-metric-arrays.tsv')
     as
     with (
-        -- sumMapState with the list of all keys with '-0.' values. Negative zero is because
-        -- sumMap removes keys with positive zeros.
+        -- sumMapState with the list of all keys with 'nan' values. 'nan' is because
```

Comment:
A simpler option is to cast the array to nullable, as `0::Nullable(Float64)` is different from the default value of Nullable(Float64) (NULL):

```
SELECT sumMap(['foo', 'foo', '4'], CAST('[-5., 5, 0]', 'Array(Nullable(Float64))')) AS t

Query id: 584265ec-d748-49c0-a4bc-4441f01f44ee

┌─t───────────────────┐
│ (['4','foo'],[0,0]) │
└─────────────────────┘

1 row in set. Elapsed: 0.001 sec.
```

Reply:
Indeed, this looks better!

Reply:
Done in https://github.com/ClickHouse/ClickHouse/pull/59333